xlml/apis/task.py

```diff
@@ -70,14 +70,91 @@ def run(self) -> DAGNode:
         group_id=self.task_test_config.benchmark_id, prefix_group_id=True
     ) as group:
       provision, queued_resource, ssh_keys = self.provision()
-      run_model = self.run_model(queued_resource, ssh_keys)
       post_process = self.post_process()
       clean_up = self.clean_up(queued_resource)
 
-      provision >> run_model >> post_process >> clean_up
+      if self.task_test_config.startup_script:
+        check_if_startup_script_end = self.check_if_startup_script_end(
+            queued_resource, ssh_keys
+        )
+        provision >> check_if_startup_script_end >> post_process >> clean_up
+      else:
+        run_model = self.run_model(queued_resource, ssh_keys)
+        provision >> run_model >> post_process >> clean_up
 
     return group
 
+  def check_if_startup_script_end(
+      self,
+      queued_resource: airflow.XComArg,
+      ssh_keys: airflow.XComArg,
+  ) -> DAGNode:
+    """Run the TPU test in `task_test_config`.
+
+    Args:
+      queued_resource: XCom value for the queued resource name (string).
+      ssh_keys: And XCom value for the TPU's SSH keys (SshKeys).
+
+    Returns:
+      A DAG node that executes the model test.
+    """
+
+    check_script = """
+# File paths
+pid_file="/tmp/main_process_id.txt"
+status_file="/tmp/process_exit_status.txt"
+
+# Check if the PID file exists
+if [ ! -f "$pid_file" ]; then
+    echo "PID file not found."
+    exit 1
+fi
+
+# Read the process ID
+pid=$(cat "$pid_file")
```

Comment:
What do you think about e.g. `tail -f --pid --retry $(cat "$pid_file") /tmp/logs` here? Tail will stream the file output until the process exits.

Not sure if we necessarily want this task to provide visibility into the output.

Reply:
Thanks! I updated to:
1. wait until `/tmp/main_process_id.txt` show up
2. Tail the log file until the process finish. 

Here is a example log:
http://shortn/_kB7XrwM1u0

Reply:
Nice! One last nit, could we move `Waiting for the workload to show up in` outside of the loop to shorten the logs some?

Reply:
Done.

Reply:
Done